### PR TITLE
Bugfix: don't fail UC model version creation if we cannot fetch its source run

### PR DESCRIPTION
**Pull request by smurching**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Fixes a bug where we currently fail UC model version creation if we cannot fetch its source run in the process of determining its workspace ID

## How is this patch tested?
Added tests, plus manually verified the fix

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

With the changes in this PR, a log line is now emitted but we don't automatically fail the creation of a model version if its source run cannot be found:
image

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/_unity_catalog/registry/rest_store.py [hunk 1]
```diff
@@ -315,7 +315,14 @@ def _get_workspace_id(self, run_id):
...
         )
-        response = verify_rest_response(response, endpoint)
+        try:
+            verify_rest_response(response, endpoint)
+        except MlflowException:
+            _logger.warning(
+                f"Unable to fetch model version's source run (with ID {run_id}) "
+                "from tracking server. The source run may be deleted or inaccessible to the "
+                "current user. No run link will be recorded for the model version."
+            )
```
Comment (harupy) [inline mlflow/store/_unity_catalog/registry/rest_store.py:325]:
```suggestion
            )
            return
```

Should we exit early here?

File: tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py [hunk 2]
```diff
@@ -321,6 +321,24 @@ def test_get_workspace_id_returns_none_if_no_request_header(store):
...
 
 
+@pytest.mark.parametrize(
+    "status_code,response_text",
+    [
+        (403, str({})),
+        (500, "<html><div>Not real json</div></html>"),
+    ],
+)
+def test_get_workspace_id_returns_none_if_request_fails(store, status_code, response_text):
```
Comment (harupy) [inline tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py:331]: Is `status_code` and `response_text` used?

File: mlflow/store/_unity_catalog/registry/rest_store.py (hunk 1 above)
Comment (smurching) [reply mlflow/store/_unity_catalog/registry/rest_store.py:325]: Yes good call to return early!

Comment (smurching) [inline mlflow/store/_unity_catalog/registry/rest_store.py:325]:
```suggestion
            )
            return None
```

File: tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py (hunk 2 above)
Comment (smurching) [inline tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py:331]:
```suggestion
def test_get_workspace_id_returns_none_if_request_fails(store):
```

**Review by harupy — APPROVED** [review]
lgtm!

**Pull request merged**